### PR TITLE
Alt-accounts finder

### PR DESCRIPTION
**Pull request by jimrandomh**
Adds an admin-only page for finding alt-accounts, linked by clientID token or IP address. This makes accessible a workflow that I've been doing with manual database queries, for checking whether new suspect accounts are alts of already-banned users.

Some parts of this were done in a pairing session with @oetherington.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1204103384282913) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx [hunk 1]
```diff
@@ -198,9 +200,9 @@ const SunshineNewUsersInfo = ({ user, classes, refetch, currentUser }: {
                 <div>
                   <UserReviewStatus user={user}/>
               
-                  {user.associatedClientId?.firstSeenReferrer && <div className={classes.qualitySignalRow}>Initial referrer: {user.associatedClientId?.firstSeenReferrer}</div>}
-                  {user.associatedClientId?.firstSeenLandingPage && <div className={classes.qualitySignalRow}>Initial landing page: {user.associatedClientId?.firstSeenLandingPage}</div>}
-                  {(user.associatedClientId?.userIds?.length??0) > 1 && <div className={classes.qualitySignalRow}>
+                  {firstClientId?.firstSeenReferrer && <div className={classes.qualitySignalRow}>Initial referrer: {firstClientId?.firstSeenReferrer}</div>}
```
Comment (Raemon) [inline packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx:203]: Make sure to migrate this into UserReviewStatus

File: packages/lesswrong/components/sunshineDashboard/UsersReviewInfoCard.tsx [hunk 2]
```diff
@@ -191,11 +194,12 @@ const UsersReviewInfoCard = ({ user, refetch, currentUser, classes }: {
         {showReviewTrigger && <MetaInfo className={classes.legacyReviewTrigger}>{reviewTrigger}</MetaInfo>}
       </div>
       <MetaInfo className={classes.referrerLandingPage}>
-        {user.associatedClientId?.firstSeenReferrer && <div>Initial referrer: {user.associatedClientId.firstSeenReferrer}</div>}
+        {firstClientId?.firstSeenReferrer && <div>Initial referrer: {firstClientId.firstSeenReferrer}</div>}
```
Comment (Raemon) [inline packages/lesswrong/components/sunshineDashboard/UsersReviewInfoCard.tsx:197]: Same here.

File: packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx (hunk 1 above)
Comment (jimrandomh) [reply packages/lesswrong/components/sunshineDashboard/SunshineNewUsersInfo.tsx]: This one showed up as a merge conflict, so was guaranteed to not be missed

File: packages/lesswrong/components/sunshineDashboard/UsersReviewInfoCard.tsx (hunk 2 above)
Comment (jimrandomh) [reply packages/lesswrong/components/sunshineDashboard/UsersReviewInfoCard.tsx:197]: This one appears to have been missed in the previous PR that attempted to refactor this block out. Refactored.

**Review by s-cheng — COMMENTED** [review]
I will let @Will-Howard do the full review, just posting the comments I wrote previously

File: packages/lesswrong/components/sunshineDashboard/ModerationAltAccounts.tsx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        setStartingIdentifier(ev.currentTarget.value);
+      }}/>
+    </div>
+    
+    <div className={classes.userInfoArea}>
+      {startingIdentifierType==="slug" && <AltAccountsNodeUserBySlug slug={startingIdentifier} classes={classes}/>}
+      {startingIdentifierType==="userId" && <AltAccountsNodeUserByID userId={startingIdentifier} classes={classes}/>}
+      {startingIdentifierType==="clientId" && <AltAccountsNodeClientID clientId={startingIdentifier} classes={classes}/>}
+      {startingIdentifierType==="ip" && <AltAccountsNodeIPAddress ipAddress={startingIdentifier} classes={classes}/>}
+    </div>
```
[the remark this answers is not in the document]
Comment (jimrandomh) [reply packages/lesswrong/components/sunshineDashboard/ModerationAltAccounts.tsx:117]: Improved somewhat (with link coloring, expander-style and hover-cursor). It's not styled to the level I'd expect for end-user UI, but it should hopefully be less confusing now.

**Review by Will-Howard — APPROVED** [review]
I used this to investigate a mod incident and it was really great, probably saved me at least an hour. 10/10 fantastic PR. Also in general I think having db grepping features like this built into the forum is good because non-dev moderators can use them which saves us some time, and they are less likely to fall into disrepair (with server scripts I often don't quite trust them)

One thing worth considering (channelling @jpaddison3) is whether we should show user _ids rather than slugs to add some friction to deanonymising people (with an extra click to show the slug once you are confident you have found something suspicious). It's easy to stumble across info like who lives with who and who works in the same office which I think we don't want to know without good cause

**Comment by jpaddison3** [comment]
> One thing worth considering (channelling @jpaddison3) is whether we should show user _ids rather than slugs to add some friction to deanonymising people (with an extra click to show the slug once you are confident you have found something suspicious). It's easy to stumble across info like who lives with who and who works in the same office which I think we don't want to know without good cause

Seems good. I have already discovered users who shared multiple IP addresses from multiple countries who turned out (whoops!) to be people who traveled together instead of alts.

File: packages/lesswrong/components/sunshineDashboard/ModerationAltAccounts.tsx [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+  
+  return <div>
+    ClientID {clientId}
+    {loading && <Loading/>}
+    {clientIdInfo && <div>
+      <div>First seen referrer: {clientIdInfo.firstSeenReferrer}</div>
+      <div>First seen landing page: {clientIdInfo.firstSeenLandingPage}</div>
+      <div>First seen: <FormatDate date={clientIdInfo.createdAt}/></div>
+      <ul>
+        {expanded /*TODO*/
```
[the remark this answers is not in the document]
Comment (jimrandomh) [reply packages/lesswrong/components/sunshineDashboard/ModerationAltAccounts.tsx:235]: It was for something I'd already done (giving this list-styles to make it clear what's clickable), and I forgot to take out the TODO.

**Comment by jimrandomh** [comment]
Changes usernames to start out as `_id`s, with a click-to-reveal.

**Pull request merged**